h/accounts/__init__.py

```diff
@@ -34,8 +35,25 @@ def make_staff(username):
         raise NoSuchUserError
 
 
+def authenticated_user(request):
+    """Return the authenticated user or None.
+
+    :rtype: h.accounts.models.User or None
+
+    """
+    if not request.authenticated_userid:
+        return None
+
+    # pylint: disable=unpacking-non-sequence
+    username, _ = util.split_user(request.authenticated_userid)
```

Comment:
Wait, I thought you wanted us to check that when `request.authenticated_userid` gets set, though? Was gonna do that separately